docs/architecture/adr-022-per-subspace-token-factory.md

```diff
@@ -0,0 +1,228 @@
+# ADR 022: Per-subspace token factory
+
+## Changelog
+- April 28th, 2023: First draft;
+
+## Status
+
+Draft
+
+## Abstract
+
+This ADR introduces a new feature that enables the subspace admins to create, mint and burn a subspace related
+coin. 
+
+## Context
+
+Desmos is a social network protocol that allows users to create, share, and engage with content 
+on a decentralized platform. 
+Currently, to post or interact with the content in a subspace you need to pay the gas fee using DSM. 
+This have the consequence of not giving any financial incentive on creating a social network on Desmos.
+
+## Decision
+
+We will wrap the Osmosis [Token Factory module](https://docs.osmosis.zone/osmosis-core/modules/tokenfactory/)
+with the following modifications:
+1. Instead of use the token creator address to compose the coin denom we will use the subspace treasury address
+   as following: `factory/{trasury_address}/subdenom`;
+2. All the operations of `CreateDenom`, `Mint`, `Burn`, `SetDenomMetadata`, `SetBeforeSendHook` and
```

Comment:
I saw `wasmd/tokenfactory` removes `SetBeforeSendHook` and `ForceTransfer`, see:
https://github.com/CosmWasm/token-factory/blob/main/x/tokenfactory/keeper/msg_server.go

As `ForceTransfer` is allowed admins to transfer tokens any users to other users forcely, and `SetBeforeSendHook` is a function to link between CW20 and tokenfactory tokens, which are not useful to our subspace tokens, both methods can be removed.

Reply:
@dadamu Is it possible to use the `SetBeforeSendHook` function to somehow implement a custom tokenomics, in your opinion? Do you see any remote possibility that this might be worth keeping anyway?

Reply:
Thanks @dadamu for the detailed explanation. I agree then we should exclude it.